docker/body_reader.go

```diff
@@ -190,7 +190,7 @@ func (br *bodyReader) Read(p []byte) (int, error) {
 			return n, fmt.Errorf("%w (after reconnecting, server did not process a Range: header, status %d)", originalErr, http.StatusOK)
 		default:
 			err := registryHTTPResponseToError(res)
-			return n, fmt.Errorf("%w (after reconnecting, fetching blob: %v)", originalErr, err)
+			return n, fmt.Errorf("%w (after reconnecting, fetching blob: %w)", originalErr, err)
```

Comment:
See `errors.Join` elsewhere — I don’t see why it’s valuable for the callers’ `errors.Is` to match against two different causes here. Only reporting the primary failure cause seems to me far more consistent.